bndtools.m2e/src/bndtools/m2e/MavenBuildRepository.java

```diff
@@ -0,0 +1,246 @@
+import aQute.bnd.service.Refreshable;
+import aQute.bnd.service.RepositoryPlugin;
+import aQute.bnd.version.Version;
+import aQute.lib.collections.SortedList;
+
+public class MavenBuildRepository extends BaseRepository implements Repository, RepositoryPlugin, Refreshable, IMavenProjectChangedListener {
+
+    private final ILogger logger = Logger.getLogger(MavenBuildRepository.class);
+
+    private final IMavenProjectRegistry mavenProjectRegistry = MavenPlugin.getMavenProjectRegistry();
+    //    private final Map<String,IMavenProjectFacade> bsnProjectFacadeMap = new HashMap<>();
+    //    private final Map<String,MavenProject> bsnMavenProjectMap = new HashMap<>();
+
+    private boolean inited = false;
+
+    private final Map<String,Entry<IMavenProjectFacade,MavenProject>> bsnMap = new HashMap<>();
+
+    @Override
+    public Map<Requirement,Collection<Capability>> findProviders(Collection< ? extends Requirement> requirements) {
+        return null;
+    }
+
+    @Override
+    public boolean refresh() throws Exception {
+        return false;
+    }
+
+    @Override
+    public File getRoot() throws Exception {
+        return null;
+    }
+
+    @Override
+    public PutResult put(InputStream stream, PutOptions options) throws Exception {
+        return null;
+    }
+
+    @Override
+    public File get(String bsn, Version version, Map<String,String> properties, DownloadListener... listeners) throws Exception {
+        if (!inited) {
+            init();
+        }
+
+        File returnVal = null;
+
+        final Entry<IMavenProjectFacade,MavenProject> entry = bsnMap.get(bsn);
+
+        if (entry != null) {
+            final MavenProject mavenProject = entry.getValue();
+
+            final File bundleFile = new File(mavenProject.getBuild().getDirectory() + "/" + mavenProject.getBuild().getFinalName() + ".jar");
+
+            if (!bundleFile.exists()) {
+                MavenUtil.execJarMojo(mavenProject, new NullProgressMonitor());
+            }
+
+            if (bundleFile.exists()) {
+                try (Jar bundle = new Jar(bundleFile)) {
+                    if (bsn.equals(bundle.getBsn()) && version.compareTo(new Version(bundle.getVersion())) == 0) {
+                        returnVal = bundleFile;
+                    }
+                }
+            }
+        }
+
+        if (returnVal != null && returnVal.exists()) {
+            if (listeners != null && listeners.length > 0) {
+                for (DownloadListener listener : listeners) {
+                    try {
+                        listener.success(returnVal);
+                    } catch (Throwable t) {}
+                }
+            }
+        }
+
+        return returnVal;
+    }
+
+    private String getBsnFromMavenProject(MavenProject mavenProject) throws Exception {
+        final File outputDir = new File(mavenProject.getBuild().getOutputDirectory());
+
+        if (outputDir.exists()) {
+            try (Jar jar = new Jar(outputDir)) {
```

Comment:
This is too expensive to just read the manifest. Use the `Domain` type around the manifest file. You only need to process the manifest.